app/src/androidTest/java/com/github/HumanLearning2021/HumanLearningApp/view/SearchTest.kt

```diff
@@ -0,0 +1,213 @@
+import androidx.test.platform.app.InstrumentationRegistry
+import com.github.HumanLearning2021.HumanLearningApp.R
+import com.github.HumanLearning2021.HumanLearningApp.TestUtils
+import com.github.HumanLearning2021.HumanLearningApp.hilt.DatabaseManagementModule
+import com.github.HumanLearning2021.HumanLearningApp.hilt.Demo2Database
+import com.github.HumanLearning2021.HumanLearningApp.hilt.DemoDatabase
+import com.github.HumanLearning2021.HumanLearningApp.hilt.ScratchDatabase
+import com.github.HumanLearning2021.HumanLearningApp.model.*
+import com.github.HumanLearning2021.HumanLearningApp.view.dataset_list_fragment.DatasetListRecyclerViewAdapter
+import com.schibsted.spain.barista.assertion.BaristaVisibilityAssertions
+import com.schibsted.spain.barista.assertion.BaristaVisibilityAssertions.assertDisplayed
+import dagger.hilt.android.testing.BindValue
+import dagger.hilt.android.testing.HiltAndroidRule
+import dagger.hilt.android.testing.HiltAndroidTest
+import dagger.hilt.android.testing.UninstallModules
+import kotlinx.coroutines.runBlocking
+import org.hamcrest.CoreMatchers
+import org.junit.*
+import org.junit.runner.RunWith
+import javax.inject.Inject
+
+@UninstallModules(DatabaseManagementModule::class)
+@HiltAndroidTest
+@RunWith(AndroidJUnit4::class)
+class SearchTest {
+
+    @get:Rule
+    val hiltRule = HiltAndroidRule(this)
+
+
+    @get:Rule
+    val activityScenarioRule: ActivityScenarioRule<MainActivity> = ActivityScenarioRule(
+        Intent(
+            ApplicationProvider.getApplicationContext(),
+            MainActivity::class.java
+        )
+    )
+
+    @BindValue
+    @Demo2Database
+    val dbManagement: DatabaseManagement = DefaultDatabaseManagement(DummyDatabaseService())
+
+    private val dummyDatasets = runBlocking { dbManagement.getDatasets() }
+
+    @Before
+    fun setup() {
+        hiltRule.inject()
+        Intents.init()
+    }
+
+    @After
+    fun cleanUp() {
+        Intents.release()
+        activityScenarioRule.scenario.close()
+    }
+
+    @Test
+    fun datasetsOverviewInputEmptyStringYieldsAllResults() {
+        navigateToDatasetsOverview()
+        onView(withId(R.id.action_search)).perform(click(), typeText("    "))
+        onView(withId(R.id.DatasetList_list)).check(
+            ViewAssertions.matches(
+                ViewMatchers.hasChildCount(dummyDatasets.size)
+            )
+        )
+    }
+
+    @Test
+    fun datasetsOverviewSearchByKeyWordYieldsCorrectResult() {
+        navigateToDatasetsOverview()
+        onView(withId(R.id.action_search)).perform(click(), typeText("kitchen"))
+        onView(withId(R.id.DatasetList_list)).check(
+            ViewAssertions.matches(
+                ViewMatchers.hasChildCount(1)
+            )
+        )
+    }
+
+    @Test
+    fun datasetsOverviewSearchNotFoundYieldsNoResult() {
+        navigateToDatasetsOverview()
+        onView(withId(R.id.action_search)).perform(click(), typeText("hi"))
```

Comment:
Maybe to make this test a bit safer, you could typeText a string that is very unlikely to appear as a dataset name, like "asdfghjkledfvboijhedfvbgbzuhikmolwsxd"

Reply:
sure thanks